### PR TITLE
added verifyBook, verifyPkgExamples for commands in docs

### PR DESCRIPTION
**Pull request by NagaHarshita**
This pr make following changes to the docs:
* Created a script `setupVerify.sh` which contains all the necessary functions to verify the output of the commands. 
* Adds mark-up for commands in the book for sections Getting Started, Concepts, Packages. 
* Adds mark-up for commands in package-examples 
* Modifies the mark-up in reference using the functions created in `setupVerify.sh`

I will add the mark-up for the rest of the sections in the book in another pr. 

Should be merged after #2030.

### CODE REVIEW

File: site/book/01-getting-started/02-quickstart.md [hunk 1]
```diff
@@ -7,22 +7,40 @@ configuration using the underlying Git version control system.
...
 `kpt pkg` commands provide the functionality for working with packages on Git and on your local
 filesystem.
 
 Next, let's quickly view the content of the package:
 
+<!-- @pkgTree @verifyGuides-->
 ```shell
-$ kpt pkg tree
-PKG: nginx
+kpt pkg tree > output.txt
```
Comment (phanimarupaka) [inline site/book/01-getting-started/02-quickstart.md:39]: This is visible to users in the docs which shouldn't be. Add extra section for each command in {{% hide %}} section to duplicate what command does and verify the output there.

File: scripts/verifyExamples.sh [hunk 2]
```diff
@@ -21,4 +21,7 @@ set -o pipefail
 mdrip -alsologtostderr -v 10 --blockTimeOut 20m0s --mode test \
     --label verifyExamples site/reference
 
+mdrip -alsologtostderr -v 10 --blockTimeOut 20m0s --mode test \
+    --label verifyGuides site/book
```
Comment (frankfarzan) [inline scripts/verifyExamples.sh:25]: verifyBook?

File: site/book/01-getting-started/02-quickstart.md [hunk 3]
```diff
@@ -7,22 +7,40 @@ configuration using the underlying Git version control system.
...
+{{% hide %}}
+
+<!-- @makeWorkplace @verifyGuides-->
+```
+# Set up workspace for the test.
+TEST_HOME=$(mktemp -d)
+cd $TEST_HOME
+touch output.txt
+
+function expectedOutput() {
```
Comment (frankfarzan) [inline site/book/01-getting-started/02-quickstart.md:19]: @phanimarupaka we should avoid redefining these functions everywhere. Any common function and repetitive set up (e.g. create tmp file) should be extracted out.

File: scripts/verifyExamples.sh (hunk 2 above)
Comment (NagaHarshita) [reply scripts/verifyExamples.sh:25]: I will change the name to verifyBook and commit them.

File: site/book/01-getting-started/02-quickstart.md (hunk 3 above)
Comment (NagaHarshita) [reply site/book/01-getting-started/02-quickstart.md]: I created setupVerify.sh script which consists of all the functions. This script is executed first in the book and references README.md files to access the functions in the remaining files. I can modify the output of the tree command once it is fixed in #2030.

File: site/book/01-getting-started/02-quickstart.md (hunk 1 above)
Comment (NagaHarshita) [reply site/book/01-getting-started/02-quickstart.md:43]: Sure, I have added extra sections as suggested.

File: package-examples/_template/README.md [hunk 4]
```diff
@@ -32,3 +32,14 @@ This step should describe the process and exact steps to render the declared val
 ### Apply the package
 
 This step should describe the process and exact commands needed to deploy the resources to a live cluster.
+
+
+{{% hide %}}
```
Comment (phanimarupaka) [inline package-examples/_template/README.md:37]: Why do you need this? It is just a template.

Comment (NagaHarshita) [reply package-examples/_template/README.md:37]: We need to first execute `source ./scripts/setupVerify.sh` to access all the functions. I placed this in README.md files of book and reference which executes first. I didn't find such a file in package-examples, but writing in template worked fine.

**Comment by mikebz** [comment]
@NagaHarshita  @phanimarupaka any way to restart this effort?

**Comment by phanimarupaka** [comment]
> @NagaHarshita @phanimarupaka any way to restart this effort?

@mikebz She did that as part of her summer project and not sure if she has time now to pick this up now. I will try to find a way to restart this project.

**Comment by mikebz** [comment]
Some of the new folks were looking for small things to pick up. Maybe post it to the chat group?

**Comment by phanimarupaka** [comment]
@sdowell Closing this PR for now. Please reopen this when you start working on it.

**Pull request closed without merging**